# Assignment 2: Data analysis - results and covid_data 

Ali Baligh-900211718 

# Content

# 1. Results data description

# 2.Variables Description 

# 3.Covid_data data description

# 4. Variables Description

# 5. Part 1a : probability of winning or losing 

# 6. Part 1b: Comparing confidence intervals.

# 7. Part 1c: Graphs

# 8. Part 1d: Description of the results.

# 9. Part 2a:  reported number of confirmed cases and deaths.

# 10. Part 2b: ratio between the confirmed cases and deaths.

# 11. Part 2c: 2020 vs 2021

# 12. Part 2d: Description of the results. 

# Results data description

In this python notebook, I will analyze the probability of winning/losing of the countries in the different tournaments  in the dataset results.csv by using statistics, visuals, and graphs. I will study the probabilities in terms of several factors (playing at home/away). I would also compare confidence intervals and draw the corresponding graphs. This data is includes 43,170 results of international football matches starting from the very first official match in 1872 up to 2019. The matches range from FIFA World Cup to FIFI Wild Cup to regular friendly matches. The matches are strictly men’s full internationals and the data does not include Olympic Games or matches where at least one of the teams was the nation’s
B-team, U-23 or a league select team. 

# Variables Description

date - date of the match

• home_team - the name of the home team

• away_team - the name of the away team

• home_score - full-time home team score including extra time, not including penaltyshootouts

• away_score - full-time away team score including extra time, not including penaltyshootouts

• tournament - the name of the tournament

• city - the name of the city/town/administrative unit where the match was played

• country - the name of the country where the match was played

• neutral - TRUE/FALSE column indicating whether the match was played at a neutral venue



# Covid_data data description

In this python notebook, I will analyze the daily reported number of cases and deaths, the case fatality rate (the ratio between deaths and cases) from COVID-19 by using statistics, visuals, and graphs. I will Compare between 2020 and 2021 and compare between regions, income, and continents). The dataset in covid_data.csv includes the records of two years 2020 and 2021 in the countries affected by the COVID-19 pandemic.

# Variables Description

• date- the date

• iso3c- the country code

• country- country name

• income- Income group according WB classification

• region- Geographical region according WB classification

• continent- the name of the continent

• dcases- the reported daily cases

• ddeaths- the reported daily deaths

• population- population size of the country in 2019

• weekdays- weekdays

• month- months


# Imports 

In [211]:
import numpy as np

In [212]:
import pandas as pd

In [213]:
from pylab import *

In [307]:
import statistics

In [296]:
import statsmodels.api as sm
from statsmodels.stats.proportion import proportion_confint

In [379]:
from pandas.api.types import CategoricalDtype

# Part 1a: probability of winning or losing

In [297]:
df=pd.read_csv('results.csv',encoding='latin-1')

In [215]:
df.head(11)

date home_team away_team  home_score  away_score tournament  \
0   1872-11-30  Scotland   England           0           0   Friendly   
1   1873-03-08   England  Scotland           4           2   Friendly   
2   1874-03-07  Scotland   England           2           1   Friendly   
3   1875-03-06   England  Scotland           2           2   Friendly   
4   1876-03-04  Scotland   England           3           0   Friendly   
5   1876-03-25  Scotland     Wales           4           0   Friendly   
6   1877-03-03   England  Scotland           1           3   Friendly   
7   1877-03-05     Wales  Scotland           0           2   Friendly   
8   1878-03-02  Scotland   England           7           2   Friendly   
9   1878-03-23  Scotland     Wales           9           0   Friendly   
10  1879-01-18   England     Wales           2           1   Friendly   

       city   country  neutral  
0   Glasgow  Scotland    False  
1    London   England    False  
2   Glasgow  Scotland    False  
3    London   England    False  
4   Glasgow  Scotland    False  
5   Glasgow  Scotland    False  
6    London   England    False  
7   Wrexham     Wales    False  
8   Glasgow  Scotland    False  
9   Glasgow  Scotland    False  
10   London   England    False

Here, the dataset is shown with 10 objects.

In order to calculate the countries probability of winning or losing, I created a data frame with some several factors that could affect the winning or lose of a country.  

In [228]:
df1=pd.DataFrame(df.home_team)

In [229]:
df1=df1.assign(away_team=df["away_team"])

In [230]:
df1=df1.assign(country=df["country"])

In [231]:
df1=df1.assign(tournament=df["tournament"])

In [232]:
goal_home=df['home_score']-df['away_score']

In [233]:
goal_away=df['home_score']-df['away_score']

In [234]:
conditions=[goal_home<0,goal_home==0,goal_home>0]
choices=['lose','draw','win']


In [235]:
conditions2=[goal_away>0,goal_away==0,goal_away<0]
choices2=['lose','draw','win']


In [236]:
df1['match_result_home'] = np.select(conditions,choices)

In [237]:
df1['match_result_away'] = np.select(conditions2,choices2)

In [238]:
df1

home_team     away_team   country              tournament  \
0          Scotland       England  Scotland                Friendly   
1           England      Scotland   England                Friendly   
2          Scotland       England  Scotland                Friendly   
3           England      Scotland   England                Friendly   
4          Scotland       England  Scotland                Friendly   
...             ...           ...       ...                     ...   
43183      Suriname        Guyana  Suriname                Friendly   
43184  Burkina Faso       Senegal  Cameroon  African Cup of Nations   
43185      Cameroon         Egypt  Cameroon  African Cup of Nations   
43186      Cameroon  Burkina Faso  Cameroon  African Cup of Nations   
43187       Senegal         Egypt  Cameroon  African Cup of Nations   

      match_result_home match_result_away  
0                  draw              draw  
1                   win              lose  
2                   win              lose  
3                  draw              draw  
4                   win              lose  
...                 ...               ...  
43183               win              lose  
43184              lose               win  
43185              draw              draw  
43186              draw              draw  
43187              draw              draw  

[43188 rows x 6 columns]

Here I created a Data frame which if a home team scores more goals, it is a win and if they score less, its a less. Same vice versa for the away team

In [239]:
df1[df1["match_result_home"]=="win"]

home_team          away_team   country              tournament  \
1       England           Scotland   England                Friendly   
2      Scotland            England  Scotland                Friendly   
4      Scotland            England  Scotland                Friendly   
5      Scotland              Wales  Scotland                Friendly   
8      Scotland            England  Scotland                Friendly   
...         ...                ...       ...                     ...   
43177     Egypt            Morocco  Cameroon  African Cup of Nations   
43178   Senegal  Equatorial Guinea  Cameroon  African Cup of Nations   
43179   Bahrain           DR Congo   Bahrain                Friendly   
43181     Nepal          Mauritius     Nepal                Friendly   
43183  Suriname             Guyana  Suriname                Friendly   

      match_result_home match_result_away  
1                   win              lose  
2                   win              lose  
4                   win              lose  
5                   win              lose  
8                   win              lose  
...                 ...               ...  
43177               win              lose  
43178               win              lose  
43179               win              lose  
43181               win              lose  
43183               win              lose  

[21009 rows x 6 columns]

Here in the data frame. It shows that from the 43188 matchs played, 21009 won if the countries were playing as a home team. So the probability of a Country winning if it was playing as a home team is 0.4865

In [240]:
df1[df1["match_result_away"]=="win"]

home_team          away_team   country              tournament  \
6           England           Scotland   England                Friendly   
7             Wales           Scotland     Wales                Friendly   
12            Wales           Scotland     Wales                Friendly   
14            Wales            England     Wales                Friendly   
16          England              Wales   England                Friendly   
...             ...                ...       ...                     ...   
43153  Sierra Leone  Equatorial Guinea  Cameroon  African Cup of Nations   
43159       Nigeria            Tunisia  Cameroon  African Cup of Nations   
43161        Guinea             Gambia  Cameroon  African Cup of Nations   
43180        Kuwait              Libya    Kuwait                Friendly   
43184  Burkina Faso            Senegal  Cameroon  African Cup of Nations   

      match_result_home match_result_away  
6                  lose               win  
7                  lose               win  
12                 lose               win  
14                 lose               win  
16                 lose               win  
...                 ...               ...  
43153              lose               win  
43159              lose               win  
43161              lose               win  
43180              lose               win  
43184              lose               win  

[12224 rows x 6 columns]

Here in the data frame. It shows that from the 43188 matchs played, 12224 won if the countries were playing as an away team. Meaning the probability of a country winning as an away team is 0.28304

# Egypt

In [241]:
df1[df1["home_team"]=="Egypt"]

home_team  away_team   country                    tournament  \
809       Egypt  Lithuania    France                      Friendly   
1463      Egypt    Hungary     Egypt                      Friendly   
1661      Egypt     Israel     Egypt  FIFA World Cup qualification   
1696      Egypt    Hungary     Italy                FIFA World Cup   
1895      Egypt     Greece     Egypt                      Friendly   
...         ...        ...       ...                           ...   
43040     Egypt      Gabon     Egypt  FIFA World Cup qualification   
43074     Egypt    Lebanon     Qatar                      Arab Cup   
43096     Egypt     Jordan     Qatar                      Arab Cup   
43148     Egypt      Sudan  Cameroon        African Cup of Nations   
43177     Egypt    Morocco  Cameroon        African Cup of Nations   

      match_result_home match_result_away  
809                 win              lose  
1463               draw              draw  
1661                win              lose  
1696               lose               win  
1895                win              lose  
...                 ...               ...  
43040               win              lose  
43074               win              lose  
43096               win              lose  
43148               win              lose  
43177               win              lose  

[393 rows x 6 columns]

In [242]:
df1[df1["away_team"]=="Egypt"]

home_team away_team      country                    tournament  \
819          Austria     Egypt      Austria                      Friendly   
824           Sweden     Egypt       Sweden                      Friendly   
1144     Netherlands     Egypt  Netherlands                      Friendly   
1146      Luxembourg     Egypt   Luxembourg                      Friendly   
1670          Israel     Egypt    Palestine  FIFA World Cup qualification   
...              ...       ...          ...                           ...   
43119        Nigeria     Egypt     Cameroon        African Cup of Nations   
43134  Guinea-Bissau     Egypt     Cameroon        African Cup of Nations   
43164    Ivory Coast     Egypt     Cameroon        African Cup of Nations   
43185       Cameroon     Egypt     Cameroon        African Cup of Nations   
43187        Senegal     Egypt     Cameroon        African Cup of Nations   

      match_result_home match_result_away  
819                 win              lose  
824                 win              lose  
1144               lose               win  
1146               draw              draw  
1670               lose               win  
...                 ...               ...  
43119               win              lose  
43134              lose               win  
43164              draw              draw  
43185              draw              draw  
43187              draw              draw  

[268 rows x 6 columns]

here it shows that Egypt played as a home team 393 times and as an away team 268 times meaning that Egypt played a total of 661 times.

In [253]:
df1[(df1["home_team"]=="Egypt")&(df1["match_result_home"]=="win")]

home_team  away_team   country                    tournament  \
809       Egypt  Lithuania    France                      Friendly   
1661      Egypt     Israel     Egypt  FIFA World Cup qualification   
1895      Egypt     Greece     Egypt                      Friendly   
3080      Egypt     Greece     Egypt                      Friendly   
3751      Egypt   Bulgaria     Egypt                      Friendly   
...         ...        ...       ...                           ...   
43040     Egypt      Gabon     Egypt  FIFA World Cup qualification   
43074     Egypt    Lebanon     Qatar                      Arab Cup   
43096     Egypt     Jordan     Qatar                      Arab Cup   
43148     Egypt      Sudan  Cameroon        African Cup of Nations   
43177     Egypt    Morocco  Cameroon        African Cup of Nations   

      match_result_home match_result_away  
809                 win              lose  
1661                win              lose  
1895                win              lose  
3080                win              lose  
3751                win              lose  
...                 ...               ...  
43040               win              lose  
43074               win              lose  
43096               win              lose  
43148               win              lose  
43177               win              lose  

[251 rows x 6 columns]

Here for example. If Egypt was playing as a home team, they won 251 matches out of the 393 played as a home team. prob to win is 0.6387

In [252]:
df1[(df1["away_team"]=="Egypt")&(df1["match_result_away"]=="win")]

home_team away_team      country  \
1144     Netherlands     Egypt  Netherlands   
1670          Israel     Egypt    Palestine   
2992          Greece     Egypt       Greece   
4101           Sudan     Egypt        Sudan   
4105           Sudan     Egypt        Sudan   
...              ...       ...          ...   
38490          Congo     Egypt        Congo   
40379       Eswatini     Egypt     Eswatini   
42874          Libya     Egypt        Libya   
43083          Sudan     Egypt        Qatar   
43134  Guinea-Bissau     Egypt     Cameroon   

                                 tournament match_result_home  \
1144                               Friendly              lose   
1670           FIFA World Cup qualification              lose   
2992                               Friendly              lose   
4101                 African Cup of Nations              lose   
4105                               Friendly              lose   
...                                     ...               ...   
38490          FIFA World Cup qualification              lose   
40379  African Cup of Nations qualification              lose   
42874          FIFA World Cup qualification              lose   
43083                              Arab Cup              lose   
43134                African Cup of Nations              lose   

      match_result_away  
1144                win  
1670                win  
2992                win  
4101                win  
4105                win  
...                 ...  
38490               win  
40379               win  
42874               win  
43083               win  
43134               win  

[75 rows x 6 columns]

and here, If Egypt was playing as an away team, they won 75 matches out of the 268 played as an away team. prob to win is 0.2799.

In [254]:
df1[(df1["home_team"]=="Egypt")&(df1["match_result_home"]=="lose")]

home_team     away_team country                            tournament  \
1696      Egypt       Hungary   Italy                        FIFA World Cup   
2986      Egypt        Turkey  Greece                              Friendly   
3425      Egypt    Yugoslavia   Egypt                              Friendly   
3587      Egypt         Italy   Egypt          FIFA World Cup qualification   
3605      Egypt       Hungary   Egypt                              Friendly   
...         ...           ...     ...                                   ...   
36654     Egypt       Senegal   Egypt  African Cup of Nations qualification   
37809     Egypt        Jordan   Egypt                              Friendly   
38765     Egypt      Cameroon   Gabon                African Cup of Nations   
40028     Egypt       Uruguay  Russia                        FIFA World Cup   
41051     Egypt  South Africa   Egypt                African Cup of Nations   

      match_result_home match_result_away  
1696               lose               win  
2986               lose               win  
3425               lose               win  
3587               lose               win  
3605               lose               win  
...                 ...               ...  
36654              lose               win  
37809              lose               win  
38765              lose               win  
40028              lose               win  
41051              lose               win  

[74 rows x 6 columns]

likewise, if Egypt was playing as a home team and still lost. They would have lost 74 times out of 393 played. meaning the prob of losing is 0.1883

In [257]:
df1[(df1["away_team"]=="Egypt")&(df1["match_result_away"]=="lose")]

home_team away_team     country                    tournament  \
819         Austria     Egypt     Austria                      Friendly   
824          Sweden     Egypt      Sweden                      Friendly   
3299         Greece     Egypt      Greece                      Friendly   
3410     Yugoslavia     Egypt  Yugoslavia                      Friendly   
3604          Italy     Egypt       Italy  FIFA World Cup qualification   
...             ...       ...         ...                           ...   
40043        Russia     Egypt      Russia                FIFA World Cup   
40060  Saudi Arabia     Egypt      Russia                FIFA World Cup   
40740       Nigeria     Egypt     Nigeria                      Friendly   
43099       Tunisia     Egypt       Qatar                      Arab Cup   
43119       Nigeria     Egypt    Cameroon        African Cup of Nations   

      match_result_home match_result_away  
819                 win              lose  
824                 win              lose  
3299                win              lose  
3410                win              lose  
3604                win              lose  
...                 ...               ...  
40043               win              lose  
40060               win              lose  
40740               win              lose  
43099               win              lose  
43119               win              lose  

[95 rows x 6 columns]

Here it shows that as an away team, they lost 95 times out of the 268 times. meaning the prob to lose is 0.3545

# With respect to different tournaments

There are many different kinds of tournaments here, to be specific 126, so I will choose the ones that I feel yield the most intresting results. 

The first one is FIFA World Cup

In [263]:
df1[((df1["home_team"]=="Egypt")^(df1["away_team"]=="Egypt"))&(df1["tournament"]=="FIFA World Cup")]

home_team away_team country      tournament  \
1696                 Egypt   Hungary   Italy  FIFA World Cup   
16216          Netherlands     Egypt   Italy  FIFA World Cup   
16228  Republic of Ireland     Egypt   Italy  FIFA World Cup   
16239              England     Egypt   Italy  FIFA World Cup   
40028                Egypt   Uruguay  Russia  FIFA World Cup   
40043               Russia     Egypt  Russia  FIFA World Cup   
40060         Saudi Arabia     Egypt  Russia  FIFA World Cup   

      match_result_home match_result_away  
1696               lose               win  
16216              draw              draw  
16228              draw              draw  
16239               win              lose  
40028              lose               win  
40043               win              lose  
40060               win              lose

Egypt played seven times in the FIFA world cup

In [274]:
df1[(df1["home_team"]=="Egypt")&(df1["tournament"]=="FIFA World Cup")&(df1["match_result_home"]=="win")]

Empty DataFrame
Columns: [home_team, away_team, country, tournament, match_result_home, match_result_away]
Index: []

Egypt did not win once as a home team in fifa as a home team

In [275]:
df1[(df1["away_team"]=="Egypt")&(df1["tournament"]=="FIFA World Cup")&(df1["match_result_away"]=="win")]

Empty DataFrame
Columns: [home_team, away_team, country, tournament, match_result_home, match_result_away]
Index: []

And Egypt also did not win once as an away team.

In [288]:
df1[(df1["home_team"]=="Egypt")&(df1["tournament"]=="Arab Cup")]

home_team     away_team country tournament match_result_home  \
15160     Egypt       Tunisia  Jordan   Arab Cup               win   
15163     Egypt       Lebanon  Jordan   Arab Cup               win   
15169     Egypt          Iraq  Jordan   Arab Cup              draw   
15171     Egypt         Syria  Jordan   Arab Cup              draw   
17270     Egypt        Jordan   Syria   Arab Cup              draw   
17287     Egypt        Kuwait   Syria   Arab Cup               win   
17295     Egypt  Saudi Arabia   Syria   Arab Cup               win   
43074     Egypt       Lebanon   Qatar   Arab Cup               win   
43096     Egypt        Jordan   Qatar   Arab Cup               win   

      match_result_away  
15160              lose  
15163              lose  
15169              draw  
15171              draw  
17270              draw  
17287              lose  
17295              lose  
43074              lose  
43096              lose

In [289]:
df1[(df1["away_team"]=="Egypt")&(df1["tournament"]=="Arab Cup")]

home_team away_team country tournament match_result_home  \
15174    Jordan     Egypt  Jordan   Arab Cup              lose   
17292     Syria     Egypt   Syria   Arab Cup              draw   
43083     Sudan     Egypt   Qatar   Arab Cup              lose   
43090   Algeria     Egypt   Qatar   Arab Cup              draw   
43099   Tunisia     Egypt   Qatar   Arab Cup               win   
43101     Qatar     Egypt   Qatar   Arab Cup              draw   

      match_result_away  
15174               win  
17292              draw  
43083               win  
43090              draw  
43099              lose  
43101              draw

Here it shows that in the Arab cup Egypt played 9 times as a home team and 6 times as a away team

In [285]:
print(df1[(df1["home_team"]=="Egypt")&(df1["tournament"]=="Arab Cup")&(df1["match_result_home"]=="win")].shape)

(6, 6)


Egypt won 6 times as a home team out of the 9 games.

In [287]:
print(df1[(df1["away_team"]=="Egypt")&(df1["tournament"]=="Arab Cup")&(df1["match_result_away"]=="win")].shape)

(2, 6)


and as an away team Egypt won 2 times.

In [290]:
df1[(df1["home_team"]=="Egypt")&(df1["tournament"]=="African Cup of Nations")]

home_team     away_team               country              tournament  \
4102      Egypt      Ethiopia                 Sudan  African Cup of Nations   
4541      Egypt      Ethiopia  United Arab Republic  African Cup of Nations   
4547      Egypt         Sudan  United Arab Republic  African Cup of Nations   
5080      Egypt        Uganda              Ethiopia  African Cup of Nations   
5547      Egypt       Nigeria                 Ghana  African Cup of Nations   
...         ...           ...                   ...                     ...   
41012     Egypt      DR Congo                 Egypt  African Cup of Nations   
41031     Egypt        Uganda                 Egypt  African Cup of Nations   
41051     Egypt  South Africa                 Egypt  African Cup of Nations   
43148     Egypt         Sudan              Cameroon  African Cup of Nations   
43177     Egypt       Morocco              Cameroon  African Cup of Nations   

      match_result_home match_result_away  
4102                win              lose  
4541                win              lose  
4547                win              lose  
5080                win              lose  
5547                win              lose  
...                 ...               ...  
41012               win              lose  
41031               win              lose  
41051              lose               win  
43148               win              lose  
43177               win              lose  

[70 rows x 6 columns]

In the African Cup of Nations, Egypt played 70 times as a home team

In [300]:
df1[(df1["away_team"]=="Egypt")&(df1["tournament"]=="African Cup of Nations")]

home_team away_team       country              tournament  \
4101           Sudan     Egypt         Sudan  African Cup of Nations   
5082        Ethiopia     Egypt      Ethiopia  African Cup of Nations   
7428        DR Congo     Egypt         Sudan  African Cup of Nations   
7433           Sudan     Egypt         Sudan  African Cup of Nations   
9721        Ethiopia     Egypt      Ethiopia  African Cup of Nations   
11299        Nigeria     Egypt       Nigeria  African Cup of Nations   
11305        Algeria     Egypt       Nigeria  African Cup of Nations   
13095       Cameroon     Egypt   Ivory Coast  African Cup of Nations   
13101    Ivory Coast     Egypt   Ivory Coast  African Cup of Nations   
13124        Algeria     Egypt   Ivory Coast  African Cup of Nations   
14964       Cameroon     Egypt       Morocco  African Cup of Nations   
16063    Ivory Coast     Egypt       Algeria  African Cup of Nations   
16068        Nigeria     Egypt       Algeria  African Cup of Nations   
16074        Algeria     Egypt       Algeria  African Cup of Nations   
18353        Nigeria     Egypt       Tunisia  African Cup of Nations   
19470       Cameroon     Egypt  South Africa  African Cup of Nations   
19485   South Africa     Egypt  South Africa  African Cup of Nations   
19491         Zambia     Egypt  South Africa  African Cup of Nations   
21226         Zambia     Egypt  Burkina Faso  African Cup of Nations   
21243    Ivory Coast     Egypt  Burkina Faso  African Cup of Nations   
21252   Burkina Faso     Egypt  Burkina Faso  African Cup of Nations   
21257   South Africa     Egypt  Burkina Faso  African Cup of Nations   
24716       Cameroon     Egypt          Mali  African Cup of Nations   
26354       Zimbabwe     Egypt       Tunisia  African Cup of Nations   
26366        Algeria     Egypt       Tunisia  African Cup of Nations   
26383       Cameroon     Egypt       Tunisia  African Cup of Nations   
30118    Ivory Coast     Egypt         Ghana  African Cup of Nations   
30123       Cameroon     Egypt         Ghana  African Cup of Nations   
32059        Algeria     Egypt        Angola  African Cup of Nations   
32062          Ghana     Egypt        Angola  African Cup of Nations   
38731           Mali     Egypt         Gabon  African Cup of Nations   
38762   Burkina Faso     Egypt         Gabon  African Cup of Nations   
43119        Nigeria     Egypt      Cameroon  African Cup of Nations   
43134  Guinea-Bissau     Egypt      Cameroon  African Cup of Nations   
43164    Ivory Coast     Egypt      Cameroon  African Cup of Nations   
43185       Cameroon     Egypt      Cameroon  African Cup of Nations   
43187        Senegal     Egypt      Cameroon  African Cup of Nations   

      match_result_home match_result_away  
4101               lose               win  
5082                win              lose  
7428               lose               win  
7433                win              lose  
9721               draw              draw  
11299               win              lose  
11305              draw              draw  
13095              lose               win  
13101              lose               win  
13124               win              lose  
14964               win              lose  
16063               win              lose  
16068               win              lose  
16074               win              lose  
18353              draw              draw  
19470               win              lose  
19485              lose               win  
19491               win              lose  
21226              lose               win  
21243              draw              draw  
21252              lose               win  
21257              lose               win  
24716               win              lose  
26354              lose               win  
26366               win              lose  
26383              draw              draw  
30118              lose               win  
30123              lose               win 

and 37 times as an away team.

In [294]:
print(df1[(df1["home_team"]=="Egypt")&(df1["tournament"]=="African Cup of Nations")&(df1["match_result_home"]=="win")].shape)

(46, 6)


Egypt won 46 times out of the 70 matches played as a home team

In [295]:
print(df1[(df1["away_team"]=="Egypt")&(df1["tournament"]=="African Cup of Nations")&(df1["match_result_away"]=="win")].shape)

(14, 6)


Egypt won 14 out of the 37 matches as the away team.

# Part 1b: Comparing confidence intervals.

# Part 1c: Graphs

# Part 1d: Description of the results.

# Part 2a:  reported number of confirmed cases and deaths.

In [301]:
df=pd.read_csv('covid_data.csv',encoding='latin-1')

In [302]:
df

date iso3c      country               income  \
0       2020-02-24   AFG  Afghanistan           Low income   
1       2020-02-25   AFG  Afghanistan           Low income   
2       2020-02-26   AFG  Afghanistan           Low income   
3       2020-02-27   AFG  Afghanistan           Low income   
4       2020-02-28   AFG  Afghanistan           Low income   
...            ...   ...          ...                  ...   
122838  2021-12-27   ZWE     Zimbabwe  Lower middle income   
122839  2021-12-28   ZWE     Zimbabwe  Lower middle income   
122840  2021-12-29   ZWE     Zimbabwe  Lower middle income   
122841  2021-12-30   ZWE     Zimbabwe  Lower middle income   
122842  2021-12-31   ZWE     Zimbabwe  Lower middle income   

                    region continent  dcases  ddeaths  population weekdays  \
0               South Asia      Asia       5        0    38041754      Mon   
1               South Asia      Asia       0        0    38041754      Tue   
2               South Asia      Asia       0        0    38041754      Wed   
3               South Asia      Asia       0        0    38041754      Thu   
4               South Asia      Asia       0        0    38041754      Fri   
...                    ...       ...     ...      ...         ...      ...   
122838  Sub-Saharan Africa    Africa    1098       17    14645468      Mon   
122839  Sub-Saharan Africa    Africa    2099       32    14645468      Tue   
122840  Sub-Saharan Africa    Africa       0        0    14645468      Wed   
122841  Sub-Saharan Africa    Africa    4180       57    14645468      Thu   
122842  Sub-Saharan Africa    Africa    1530        7    14645468      Fri   

       month  
0        Feb  
1        Feb  
2        Feb  
3        Feb  
4        Feb  
...      ...  
122838   Dec  
122839   Dec  
122840   Dec  
122841   Dec  
122842   Dec  

[122843 rows x 11 columns]

In [313]:
statistics.mean(df["dcases"])

2353.722996019309

In [315]:
statistics.median(df["dcases"])

98

In [314]:
statistics.stdev(df["dcases"])

12256.414468539499

The mean of the daily reported cases is 2353.72, however it is so spread out since the standard deviation is bigger than the mean and the median is 98 which is much smaller than the mean. 

In [316]:
statistics.mean(df["ddeaths"])

44.293024429556425

In [317]:
statistics.median(df["ddeaths"])

1

In [318]:
statistics.stdev(df["ddeaths"])

197.94510823044232

For the daily confirmed deaths, it's mean is 44.29 with a median of 1. It's standard deviation is also high with it being bigger than the mean at 197.95.

In [341]:
df[df["dcases"]==max(df["dcases"])]

date iso3c country               income                 region  \
112999  2020-12-10   TUR  Turkey  Upper middle income  Europe & Central Asia   

       continent  dcases  ddeaths  population weekdays month  
112999      Asia  823225      220    83429615      Thu   Dec

The country with the biggest amount of confirmed cases in a day which is 823225 was Turkey in 2020-12-10 in Thursday. 

In [345]:
df[df["ddeaths"]==max(df["ddeaths"])]

date iso3c  country               income  \
33884  2021-07-20   ECU  Ecuador  Upper middle income   

                          region                 continent  dcases  ddeaths  \
33884  Latin America & Caribbean  South America(continent)    1634     8786   

       population weekdays month  
33884    17373662      Tue   Jul

The country with the biggest amount of confirmed deaths in a day which is 823225 was Ecuador in 2021-07-20 in Tuesday. 

In [378]:
cats=['Fri', 'Sat', 'Sun','Mon','Tue','Wed','Thu']
cat_type = CategoricalDtype(categories=cats, ordered=True)
df['weekdays'] = df['weekdays'].astype(cat_type)

In [377]:
df.groupby("weekdays").agg({"dcases": [np.mean, np.std, np.size]})

dcases                     
                 mean           std   size
weekdays                                  
Fri       2643.215039  13626.207863  17634
Sat       2174.297447  10922.667935  17472
Sun       1852.340706   9559.968071  17496
Mon       2116.600879  11700.105447  17521
Tue       2388.510716  11756.420050  17544
Wed       2601.990272  12848.605924  17578
Thu       2693.674338  14618.890740  17598

In [381]:
df.groupby("weekdays").agg({"ddeaths": [np.mean, np.std, np.size]})

ddeaths                   
               mean         std   size
weekdays                              
Fri       47.990756  208.404740  17634
Sat       40.258413  177.949619  17472
Sun       31.856367  139.320870  17496
Mon       37.954968  155.752300  17521
Tue       50.840002  230.863861  17544
Wed       51.346968  226.105877  17578
Thu       49.695534  224.874945  17598

Here it shows all the weekdays of confirmed daily cases and deaths, it shows it's mean, standard deviation and the size.

The day with the biggest mean is Thursday while the lowest is Sunday for daily confirmed cases and deaths

In [ ]:
cats=['Jan', 'Feb', 'Mar','Apr','May','Jun','Jul','Aug','Sep','Oct','Nov','Dec']
cat_type = CategoricalDtype(categories=cats, ordered=True)
df['month'] = df['month'].astype(cat_type)

In [380]:
df.groupby("month").agg({"dcases": [np.mean, np.std, np.size]})

dcases                     
              mean           std   size
month                                  
Jan    3330.493183  16114.132615   5868
Feb    1869.261968   7579.872208   6058
Mar    1599.375677   6590.844635   9785
Apr    2294.984731  14842.094035  10872
May    1989.523264  16428.338523  11348
Jun    1433.149046   6561.947273  11010
Jul    2007.870352   7408.384725  11377
Aug    2441.502152  10425.483877  11387
Sep    2223.283062  10260.530149  11040
Oct    2205.899580   7917.525038  11432
Nov    2964.868421  11235.400964  11134
Dec    3979.827003  20864.089717  11532

In [382]:
df.groupby("month").agg({"ddeaths": [np.mean, np.std, np.size]})

ddeaths                   
            mean         std   size
month                              
Jan    71.127812  289.935659   5868
Feb    51.762133  222.814231   6058
Mar    34.963413  171.864116   9785
Apr    52.986479  240.192244  10872
May    46.761720  254.388290  11348
Jun    38.822434  201.438666  11010
Jul    39.192142  175.788522  11377
Aug    42.456749  157.551906  11387
Sep    39.211051  168.449391  11040
Oct    35.094822  143.726925  11432
Nov    44.249506  150.833802  11134
Dec    50.099723  203.325567  11532

Here it shows all the months of confirmed daily cases and deaths, it shows it's mean, standard deviation and the size.

The month with the biggest mean is December while the lowest is June for daily confirmed cases and for deaths, the biggest mean is january while the lowest is March. 

# Part 2b: ratio between the confirmed cases and deaths.

According to the median, for every 98 confirmed cases, there was 1 death. 

# Part 2c: 2020 vs 2021

# Part 2d: Description of the results